nextcord/ext/commands/bot.py

```diff
@@ -698,7 +698,10 @@ def _load_from_module_spec(
 
         extras = extras or {}
         try:
-            setup(self, **extras)
+            if inspect.iscoroutinefunction(setup):
+                asyncio.run(setup(self, **extras))
```

Comment:
Asyncio.run is used as a program entry point, and should ideally only be called once during a programs lifecycle. Therefore using it for each cog is not the go

https://docs.python.org/3/library/asyncio-task.html#asyncio.run

Reply:
how do you think about making _load_from_module_specs a coroutine?